spark/src/main/scala/org/apache/comet/serde/comparisons.scala

```diff
@@ -0,0 +1,265 @@
+  override def convert(
+      expr: Expression,
+      inputs: Seq[Attribute],
+      binding: Boolean): Option[ExprOuterClass.Expr] = {
+    val inExpr = expr.asInstanceOf[In]
+    in(expr, inExpr.value, inExpr.list, inputs, binding, negate = false)
+  }
+}
+
+sealed trait ComparisonBase {
+
+  /**
+   * Creates a UnaryExpr by calling exprToProtoInternal for the provided child expression and then
+   * invokes the supplied function to wrap this UnaryExpr in a top-level Expr.
+   *
+   * @param child
+   *   Spark expression
+   * @param inputs
+   *   Inputs to the expression
+   * @param f
+   *   Function that accepts an Expr.Builder and a UnaryExpr and builds the specific top-level
+   *   Expr
+   * @return
+   *   Some(Expr) or None if not supported
+   */
+  def createUnaryExpr(
+      expr: Expression,
+      child: Expression,
+      inputs: Seq[Attribute],
+      binding: Boolean,
+      f: (ExprOuterClass.Expr.Builder, ExprOuterClass.UnaryExpr) => ExprOuterClass.Expr.Builder)
+      : Option[ExprOuterClass.Expr] = {
+    val childExpr = exprToProtoInternal(child, inputs, binding) // TODO review
+    if (childExpr.isDefined) {
+      // create the generic UnaryExpr message
+      val inner = ExprOuterClass.UnaryExpr
+        .newBuilder()
+        .setChild(childExpr.get)
+        .build()
+      // call the user-supplied function to wrap UnaryExpr in a top-level Expr
+      // such as Expr.IsNull or Expr.IsNotNull
+      Some(
+        f(
+          ExprOuterClass.Expr
+            .newBuilder(),
+          inner).build())
+    } else {
+      withInfo(expr, child)
+      None
+    }
+  }
+
+  def createBinaryExpr(
+      expr: Expression,
+      left: Expression,
+      right: Expression,
+      inputs: Seq[Attribute],
+      binding: Boolean,
+      f: (ExprOuterClass.Expr.Builder, ExprOuterClass.BinaryExpr) => ExprOuterClass.Expr.Builder)
+      : Option[ExprOuterClass.Expr] = {
+    val leftExpr = exprToProtoInternal(left, inputs, binding)
+    val rightExpr = exprToProtoInternal(right, inputs, binding)
+    if (leftExpr.isDefined && rightExpr.isDefined) {
+      // create the generic BinaryExpr message
+      val inner = ExprOuterClass.BinaryExpr
+        .newBuilder()
+        .setLeft(leftExpr.get)
+        .setRight(rightExpr.get)
+        .build()
+      // call the user-supplied function to wrap BinaryExpr in a top-level Expr
+      // such as Expr.And or Expr.Or
+      Some(
+        f(
+          ExprOuterClass.Expr
+            .newBuilder(),
+          inner).build())
+    } else {
+      withInfo(expr, left, right)
+      None
+    }
+  }
+
+  // Utility method. Adds explain info if the result of calling exprToProto is None
+  def optExprWithInfo(
+      optExpr: Option[Expr],
+      expr: Expression,
+      childExpr: Expression*): Option[Expr] = {
+    optExpr match {
+      case None =>
+        withInfo(expr, childExpr: _*)
+        None
+      case o => o
+    }
+  }
+
+  def in(
```

Comment:
This is only called in one location. Can we move this logic to the call site in `CometIn` since it seems specific to that expression?